frontend/src/lib/components/ObjectTags/ObjectTags.tsx

```diff
@@ -70,89 +74,123 @@ export function ObjectTags({
-                      .filter((t) => !!t)
-                      .map((tag, index) => {
-                          return (
-                              <Tag
-                                  key={index}
-                                  color={COLOR_OVERRIDES[tag] || colorForString(tag)}
-                                  style={{ marginTop: 8 }}
-                              >
-                                  {tag}{' '}
-                                  {!staticOnly &&
-                                      onChange &&
-                                      (deletedTags.includes(tag) ? (
-                                          <SyncOutlined spin />
-                                      ) : (
-                                          <CloseOutlined
-                                              className="click-outside-block"
-                                              style={{ cursor: 'pointer' }}
-                                              onClick={() => handleDelete(tag)}
-                                          />
-                                      ))}
-                              </Tag>
-                          )
-                      })}
-            {!staticOnly && onChange && saving !== undefined && (
-                <span style={{ display: 'inline-flex', fontWeight: 400 }}>
-                    <Tag
-                        onClick={() => setAddingNewTag(true)}
-                        data-attr="button-add-tag"
-                        style={{
-                            cursor: 'pointer',
-                            borderStyle: 'dashed',
-                            backgroundColor: '#ffffff',
-                            display: addingNewTag ? 'none' : 'initial',
-                        }}
-                        icon={<PlusOutlined />}
-                    >
-                        Add tag
-                    </Tag>
-                    {addingNewTag && (
-                        <SelectGradientOverflow
-                            size="small"
-                            onBlur={() => setAddingNewTag(false)}
-                            data-attr="new-tag-input"
-                            autoFocus
-                            allowClear
-                            autoClearSearchValue
-                            defaultOpen
-                            showSearch
-                            style={{ width: 160 }}
-                            onChange={handleAdd}
-                            loading={saving}
-                            onSearch={setNewTag}
-                            placeholder='try "official"'
+        <div style={{ display: 'flex', flexWrap: 'wrap', gap: 8, ...style }} className={className} data-attr={dataAttr}>
+            {paywall ? (
+                <Tooltip
+                    title={
+                        paywall
+                            ? "This field is part of PostHog's tagging collaboration set and requires a premium plan."
```

Comment:
The language here is a bit awkward. Maybe it should be "Creating tags requires a premium PostHog plan"

Reply:
Related thought, maybe we should use the `UpgradeModal` here? I think this is exactly what it's meant for.

Reply:
Nice UpgradeModal is pretty slick, didn't know that existed.